vernac/metasyntax.ml

```diff
@@ -665,15 +665,28 @@ let expand_list_rule s typ tkl x n p ll =
        aux (i+1) (main :: tks @ hds) ll in
   aux 0 [] ll
 
-let is_constr_typ typ x etyps =
+let production_level_eq lev1 lev2 =
```

Comment:
Also, sorry to nitpick, but this should go to Ppextend. There are already a few equality functions on Extend types there, better to centralize them.

Reply:
`Ppextend` is more for pretty-printing. An alternative could be to add it to `extend.ml` and create `extend.mli`. Would it be ok for you?

About automatic generation of eq schemes, did someone already looked at ppx_deriving?

Reply:
Fine with defining it into the Extend module. It's a crime to have ml-only files anyways. Regarding PPX, I'd prefer not to rely on it as long as possible. It's been plagued with backwards compatibility issues and it looks more like a *usine à gaz* to me. In an ideal world the OCaml language would provide native deriving facilities as e.g. in Rust .

Reply:
> Fine with defining it into the Extend module. It's a crime to have ml-only files anyways.

OK, I moved it to `extend.ml` and added a mli. I uniformized the indentation on the way. (Don't hesitate to push to the PR a different uniformization of indentation if you prefer.)

I assume that the change is ok also to @ejgallego. In any case, it shall still possible to think at different organization of files and contents in the future. (And I have actually myself other ongoing changes around notations.)

> In an ideal world the OCaml language would provide native deriving facilities as e.g. in Rust.

Maybe should we talk to an ocaml developer about that...